detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/naming/MemberNameEqualsClassName.kt

```diff
@@ -82,22 +80,20 @@ class MemberNameEqualsClassName(config: Config = Config.empty) : Rule(config) {
         super.visitObjectDeclaration(declaration)
     }
 
-    private fun getMisnamedMembers(klassOrObject: KtClassOrObject, name: String?): List<KtNamedDeclaration> {
-        val body = klassOrObject.body ?: return emptyList()
-        val declarations = getFunctions(body).concat(body.properties)
-        return declarations?.filter { it.name?.equals(name, ignoreCase = true) == true }.orEmpty()
+    private fun getMisnamedMembers(klassOrObject: KtClassOrObject, name: String?): Sequence<KtNamedDeclaration> {
+        val body = klassOrObject.body ?: return emptySequence()
+        return (getFunctions(body) + body.properties)
```

Comment:
```suggestion
        return (body.functions + body.properties)
```

There's a `functions` property on `KtClassBody` which might be used instead of the private `getFunctions` function?

https://github.com/JetBrains/kotlin/blob/v1.3.60/compiler/psi/src/org/jetbrains/kotlin/psi/KtClassBody.kt#L46-L47

Reply:
Done